src/fluree/db/nameservice/storage.cljc

```diff
@@ -77,29 +115,47 @@
 
   (all-records [_]
     (go-try
-      ;; Use recursive listing to support ledger names with '/' characters
-      (if (satisfies? storage/RecursiveListableStore store)
-        (if-let [list-paths-result (storage/list-paths-recursive store "ns@v1")]
-          (loop [remaining-paths (<? list-paths-result)
-                 records []]
-            (if-let [path (first remaining-paths)]
-              (let [file-content (<? (storage/read-bytes store path))]
-                (if file-content
-                  (let [content-str (if (string? file-content)
-                                      file-content
-                                      #?(:clj (let [^bytes bytes-content file-content]
-                                                (String. bytes-content "UTF-8"))
-                                         :cljs (js/String.fromCharCode.apply nil file-content)))
-                        record (json/parse content-str false)]
-                    (recur (rest remaining-paths) (conj records record)))
-                  (recur (rest remaining-paths) records)))
-              records))
-          [])
-        ;; Fallback for stores that don't support ListableStore
-        (do
-          (log/warn "Storage backend does not support RecursiveListableStore protocol")
+      ;; Read all JSON files from ns@v1 directory structure
+      (let [base-dir "ns@v1/"
+            all-files (<? (storage/list-paths-recursive store base-dir))]
+        ;; Use pipeline for parallel reading and parsing
+        (if (seq all-files)
+          (let [input-ch (async/chan 100 (filter #(str/ends-with? % ".json")))
+                _ (async/onto-chan! input-ch all-files)
+                output-ch (async/chan 100)
+                ;; Function to read and parse a single file
+                read-and-parse (fn [file-path out-ch]
+                                 (go
+                                   (try
+                                     (when-let [record-bytes (<? (storage/read-bytes store file-path))]
+                                       (let [record (json/parse record-bytes false)]
+                                         (async/>! out-ch record)))
+                                     #?(:clj (catch Exception e
```

Comment:
I think any errors should be propagated to the calling function. A user has to be watching the logs to know something went wrong as opposed to there being less or no records.

Reply:
This had already been refactored previously.